### PR TITLE
[AMD] Enable preshuffle paged MQA and page_size=64 for NSA indexer

### PR DESCRIPTION
**Pull request by 1am9trash**
> Depends on aiter [PR#2879](https://github.com/ROCm/aiter/pull/2879) — must be merged first.

## Motivation

In the NSA indexer, paged mqa computation is very slow at high concurrency. For example, at 8k1kcc64, it takes ~88 us per layer, consisting of two kernels (logits tensor init 11us + MQA kernel 77us).

image

The main bottleneck is the oversized `block_table`. The `block_table` shape is (batch_size, max_seq_len / page_size). With `page_size=1` and `max_seq_len=131k` at cc64, the `block_table` shape is (64, 131072), totaling 64 × 131072 × 4B ≈ 32MB, with frequent indirect loads, causing poor MQA kernel performance.

## Modifications

To reduce the block_table size, we cannot use `page_size=1`. In this PR, we change it to 64. This introduces the following changes:
- The original MQA kernel does not support `page_size>1`. We switch to the MQA preshuffle kernel, which supports `page_size` as a multiple of 16.
- Since the MQA preshuffle kernel requires a specific k-cache layout, the indexer's k-cache read/write paths must also store data accordingly. This was not supported in the existing kernels. In aiter PR#2879, we add preshuffle support to the indexer k-cache read/write kernels (controlled by `preshuffle=True`).

Other changes:
- Replace `torch.full(..., -inf)` with `torch.empty(...)` to eliminate an unnecessary initialization kernel.
- In this PR, we also update `page_size` initialization and corresponding assertions.

## Accuracy Tests

<details>
<summary>GLM-5.1-FP8 launch cmd</summary>

```bash
export SGLANG_ROCM_FUSED_DECODE_MLA=0
export ROCM_QUICK_REDUCE_QUANTIZATION=INT4
export SAFETENSORS_FAST_GPU=1
python3 -m sglang.launch_server \
  --model-path GLM-5.1-FP8 \
  --tp 8 --port 9000 --trust-remote-code \
  --tool-call-parser glm47 --reasoning-parser glm45 \
  --mem-fraction-static 0.85 \
  --model-loader-extra-config '{"enable_multithread_load": true, "num_threads": 8}' \
  --nsa-prefill-backend tilelang --nsa-decode-backend tilelang --disable-radix-cache \
  --kv-cache-dtype fp8_e4m3
```
</details>

MI355 GSM8k (TP8): 0.951

## Speed Tests and Profiling

Per-layer profiling:
- Kernels: 2 -> 1

Critical case  (8k1kcc64):
- Time per layer: 88us -> 19us
- e2e +16% throughput.
image

Benchmark on MI355X TP8, concurrency 4/8/16/32/64 averaged:
- ISL/OSL 1k/1k: Throughput +3.83%, TPOT -3.26%
- ISL/OSL 8k/1k: Throughput +5.61%, TPOT -6.65%

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

File: python/sglang/srt/layers/attention/nsa/nsa_indexer.py [hunk 1]
```diff
@@ -506,7 +501,7 @@ def _get_topk_paged(
                 seqlens_32,
                 block_tables,
                 max_seq_len,
-                Preshuffle=False,
+                Preshuffle=True,
```
Comment (Jacob0226) [inline python/sglang/srt/layers/attention/nsa/nsa_indexer.py]: Should we replace `True` with `_use_aiter`? Since the provider code on line 1038 is gated in `_use_aiter`.

Comment (1am9trash) [reply python/sglang/srt/layers/attention/nsa/nsa_indexer.py]: Addressed. Thanks.

**Comment by Jacob0226** [comment]
LGTM

**Review by HaiShaw — APPROVED** [review]

**Comment by HaiShaw** [comment]
@amd-bot ci-status
